### ISSUE TITLE
Invalid `__slots__` class variable defined (Updated)

### ISSUE THREAD

**Issue by nait-aul** [PyCQA/pylint#1130]
### Steps to reproduce

```python
class Foo:
    __slots__ = attribute_name

    def __init__(self, value: str):
        self.attribute_name = value
```

### Current behavior

```bash
E:  2,16: Undefined variable 'attribute_name' (undefined-variable)
```

### Expected behavior

Missing quotation marks (more polite description). Should be:

```python
class Foo:
    __slots__ = "attribute_name"

    ...
```

[Official documentation](https://docs.python.org/3/reference/datamodel.html#object.__slots__) says:

> The `__slots__` class variable can be assigned a string, iterable, or sequence of strings with variable names used by instances.

### pylint --version output

```bash
pylint 1.6.4
```

**Comment by AWhetter** [comment]
When running with master I get a message for "`attribute_name`" not being defined. I think this is what it should do rather than ask for quotation marks considering that `__slots__` can be an iterable.

Would you agree? Is this behaviour that you are looking for?

**Comment by nait-aul** [comment]
Ahh, forget to mention, that the `attribute_name` is an attribute name defined in the `__init__` method and it's not an interator, but let's say a string value. That's what I was looking for. Will edit my previous post.

**Comment by AWhetter** [comment]
Even with the new test case I get an "`Undefined variable 'attribute_name'`" message when running with master.

**Comment by nait-aul** [comment]
Well, when I think about it now, it would be more polite to "shout" missing quation marks than `undifined variable` if the attribute exists in the `__init__`.

What do you think, bad idea or keep it as it is?

**Comment by AWhetter** [comment]
Considering that `__slots__` can be an iterable or a string, saying that it should have quotation marks could be misleading. So I think we should leave it as is.

**Comment by nait-aul** [comment]
OK, I'm gonna close this issue.
